syntax = "proto3";
option cc_generic_services = true;

import "common.proto";

package inbox; 

message InboxItemInfo
{
   common.UniqueId id = 1;
   string text = 2;
}

message InboxItems
{
   repeated InboxItemInfo items = 1;
}

service InboxService 
{
   rpc GetInbox (common.EmptyMessage) returns (InboxItems);
   rpc DeleteItem (common.UniqueId) returns (common.OperationResultMessage);
   rpc EditItem (InboxItemInfo) returns (common.OperationResultMessage);
   rpc AddItem (InboxItemInfo) returns (common.UniqueId);
}

